cli/cfncluster/cfnconfig.py

```diff
@@ -268,47 +271,122 @@ def __init__(self, args):
-                                                % self.__cluster_section)
+                      % self.__cluster_section)
                 sys.exit(1)
-            self.__ebs_section = ('ebs %s' % self.__ebs_settings)
+            #modify list
+            self.__ebs_section = str(self.__ebs_settings).split(',')
+            if(len(self.__ebs_section) > self.__MAX_EBS_VOLUMES):
+                print("ERROR: number of EBS volumes requested is greater than the MAX.\n"
+                      "Max number of EBS volumes supported is currently %s" %self.__MAX_EBS_VOLUMES)
+                sys.exit(1)
+            self.parameters.append(("NumberOfEBSVol", str(len(self.__ebs_section))))
+            for i, item in enumerate(self.__ebs_section):
+                item = ('ebs %s' % item.strip())
+                self.__ebs_section[i] = item
         except configparser.NoOptionError:
             pass
 
-        # Dictionary list of all EBS options
-        self.__ebs_options = dict(ebs_snapshot_id=('EBSSnapshotId','EC2Snapshot'), volume_type=('VolumeType',None),
-                                  volume_size=('VolumeSize',None), ebs_kms_key_id=('EBSKMSKeyId', None),
-                                  volume_iops=('VolumeIOPS',None), encrypted=('EBSEncryption',None),
-                                  ebs_volume_id=('EBSVolumeId','EC2Volume')
-                                  )
-
+        # Handle the shared_dir under EBS setting sections
+        __temp_dir_list = []
         try:
             if self.__ebs_section:
-                for key in self.__ebs_options:
+                for section in self.__ebs_section:
                     try:
-                        __temp__ = __config.get(self.__ebs_section, key)
+                        __temp__ = __config.get(section, 'mount_point')
                         if not __temp__:
                             print("ERROR: %s defined but not set in [%s] section"
-                                                    % (key, self.__ebs_section))
+                                  % (key, section))
                             sys.exit(1)
-                        if self.__sanity_check and self.__ebs_options.get(key)[1] is not None:
-                            config_sanity.check_resource(self.region,self.aws_access_key_id, self.aws_secret_access_key,
-                                                self.__ebs_options.get(key)[1],__temp__)
-                        self.parameters.append((self.__ebs_options.get(key)[0],__temp__))
+                        __temp_dir_list.append(__temp__)
+
                     except configparser.NoOptionError:
                         pass
+                    except configparser.NoSectionError:
+                        print("ERROR: [%s] section defined in [ebs_settings] does not exist" %section)
+                        sys.exit(1)
+        except AttributeError:
+            pass
+
+        try:
+            if len(__temp_dir_list) == len(self.__ebs_section):
```

Comment:
What if both the lists are empty?

After talking to Rex, there is no case where __ebs_section can exist as an empty list.